iceoryx2-bb/posix/tests/periodic_timer_tests.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+mod periodic_timer {
+    use iceoryx2_bb_posix::periodic_timer::*;
+    use iceoryx2_bb_testing::assert_that;
+    use std::time::Duration;
+
+    #[test]
+    fn next_iteration_works_smallest_timeout_added_first() {
```

Comment:
I completely agree, but this virtual or simulation clock must then be consistently used, in the full stack to avoid a mixture of a simulation and a real clock. My idea was to introduce a testing platform that provides exactly this kind of clock on the lowest level.

But we do not yet have this. So I used large and tiny timeouts to not run into flakyness.